src/main/java/graphql/testrunner/service/GitService.java

```diff
@@ -0,0 +1,34 @@
+package graphql.testrunner.service;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import graphql.testrunner.util.TestRunnerException;
+
+@Service
+public class GitService {
+
+    private static final Logger LOGGER = Logger.getLogger(GitService.class.getName());
+    private static final String BRANCH = "new-branch";
+
+    @Autowired
+    private Git git;
+
+    public void checkout(String commitHash) throws TestRunnerException {
+        try {
+            git.checkout()
+              .setCreateBranch(true)
+              .setName(BRANCH)
```

Comment:
Is it safe to use the same branch name for every run? I know we're not expecting to support concurrent runs, but Bad Things Do Happen and if Git is backed by files I imagine this could maybe cause problems.

Do we even need a branch at all?

Reply:
Yes, Jgit throws error if we don't pass the branch name. We will append unique jobid to the branch name in future stories and delete the branch as part of cleanup.